### PR TITLE
Add CentOS build check to CI

### PR DESCRIPTION
**Pull request by wenzeslaus**
This uses Docker to run CentOS 7 and then conda to get Python 3 and Python packages.
There is only the simple test running and the packages and linking to dependencies
is not optimal.

### CODE REVIEW

**Comment by wenzeslaus** [comment]
@neteler I hope you find this helpful (originally I was thinking just email, but then PR seemed more useful in the long run). However, this is about what I can do here. I combined my HPC instructions with your wiki instructions and got this. If you want to polish it or take it further feel free, but I'm fine with merging this as is. If nothing else, it is useful at least for capturing dependencies needed.

**Comment by neteler** [comment]
Cool, thanks!
The other day I managed to get the SPEC file right for EPEL7:
https://src.fedoraproject.org/rpms/grass/blob/master/f/grass.spec

I'll compare that for the review.

File: .github/workflows/build_centos.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+./configure \
+    --prefix="$INSTALL_PREFIX/" \
+    --without-freetype \
+    --with-sqlite \
+    --with-cairo --with-cairo-ldflags=-lfontconfig \
+    --with-freetype --with-freetype-includes=/usr/include/freetype2 \
+    --with-proj --with-proj-share=/usr/share/proj \
+    --with-gdal=/usr/bin/gdal-config \
+    --without-zstd \
```
Comment (neteler) [inline .github/workflows/build_centos.sh:33]: Here ZSTD support could be enabled:

EPEL7: zstd-1.4.4-1.el7.x86_64.rpm

File: .github/workflows/yum.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+proj-epsg
+proj-nad
+python-dateutil
+python-devel
+python-imaging
+python-matplotlib
+sqlite-devel
+wxGTK-devel
+wxPython
+zlib-devel
```
Comment (neteler) [inline .github/workflows/yum.txt:37]:
```suggestion
zlib-devel
zstd
```
Here ZSTD support could be enabled:

EPEL7: zstd-1.4.4-1.el7.x86_64.rpm

**Comment by neteler** [comment]
Thanks for setting this up. The current list is:

```
  BLAS support:               no
  BZIP2 support:              no
  C++ support:                yes
  Cairo support:              yes
  DWG support:                no
  FFTW support:               no
  FreeType support:           yes
  GDAL support:               yes
  GEOS support:               no
  LAPACK support:             no
  Large File support (LFS):   yes
  libLAS support:             no
  MySQL support:              no
  NetCDF support:             no
  NLS support:                no
  ODBC support:               no
  OGR support:                yes
  OpenCL support:             no
  OpenGL support:             yes
  OpenMP support:             no
  PDAL support:               no
  PNG support:                yes
  POSIX thread support:       no
  PostgreSQL support:         no
  Readline support:           no
  Regex support:              yes
  SQLite support:             yes
  TIFF support:               no
  X11 support:                yes
  Zstandard support:          no

  using old PROJ 4 API
```

- Zstandard support could be easily enabled, see above
- only Centos 8 comes with a more recent PROJ 6

Overall, it looks good to me.

**Comment by wenzeslaus** [comment]
The zstd change failed, so I'm reverting it and merging in the original state.

Here is the error for reference (and a future PR):

```
2020-05-12T19:54:13.4330180Z checking whether to use zstd... yes
2020-05-12T19:54:13.4334757Z checking for location of zstd includes... 
2020-05-12T19:54:13.4492187Z configure: error: *** Unable to locate zstd includes.
2020-05-12T19:54:13.4492363Z checking for zstd.h... no
```

**Comment by neteler** [comment]
> The zstd change failed, so I'm reverting it and merging in the original state.
> 
> Here is the error for reference (and a future PR):
> 
> ```
> 2020-05-12T19:54:13.4330180Z checking whether to use zstd... yes
> 2020-05-12T19:54:13.4334757Z checking for location of zstd includes... 
> 2020-05-12T19:54:13.4492187Z configure: error: *** Unable to locate zstd includes.
> 2020-05-12T19:54:13.4492363Z checking for zstd.h... no
> ```

Yes, strangely not there. I am building it on COPR (https://copr.fedorainfracloud.org/coprs/neteler/zstd/) now and hope it shows up in Centos 8 soon.

**Pull request merged**